blockstore.go

```diff
@@ -183,6 +186,21 @@ func (bs *blockstore) Has(k *cid.Cid) (bool, error) {
 	return bs.datastore.Has(dshelp.CidToDsKey(k))
 }
 
+func (bs *blockstore) GetSize(k *cid.Cid) (int, error) {
+	maybeData, err := bs.datastore.Get(dshelp.CidToDsKey(k))
+	if err == ds.ErrNotFound {
+		return -1, nil
```

Comment:
Taking a look... I'll need to think about this. 

Removing that check on Line 191: Triggers three test failures:
```Go
--- FAIL: TestPutManyAddsToBloom (0.00s)
	bloom_cache_test.go:68: datastore: key not found
--- FAIL: TestGetAndDeleteFalseShortCircuit (0.00s)
	arc_cache_test.go:41: get hit datastore
--- FAIL: TestHasRequestTriggersCache (0.00s)
	arc_cache_test.go:41: has hit datastore
```

The two arc_cache_test failures are because:
1) The test case assumes no errors when a block is not found via 'arccache.Has'. Removing the check in 'GetSize/Line 191' triggers this failure because arccache.Has has a dependency on GetSize: 
```Go 
func (b *arccache) Has(k *cid.Cid) (bool, error) {
	blockSize, err := b.GetSize(k)
	return blockSize > -1, err
}
```

The bloom_cache_test failure is due to similar reason:
1) The test case assumes no errors when a block is not found via 'bloomcache.GetSize'.
```Go 
       blockSize, err = cachedbs.GetSize(block2.Cid())
	if err != nil {
		t.Fatal(err) //<--Test fails here bloom_cache_test.go:68: datastore: key not found
	}
	if blockSize > -1 || has {
		t.Fatal("not added block is reported to be in blockstore")
	}
```

This is what I need to think about a bit more... 
**Open Question:**
Ideally, should 'Has' be dependent on 'GetSize'? or decoupled? 

**Open Question:** 
Ideally, should 'GetSize' and/or 'Has' return an error when a block is not found? The current implementation for 'Has' w/ or w/o my changes currently suppresses the error in this case.

Reply:
***Solution***
I can pull that check out of 'GetSize' on line 191 and add it to 'arccache.Has'
1) This would allow for 'Has' to return false instead of an error when a block isn't found.
2) This would also allow 'GetSize' to return bs.ErrNotFound when a block isn't found

```Go
func (b *arccache) Has(k *cid.Cid) (bool, error) {
	blockSize, err := b.GetSize(k)
	if err == ds.ErrNotFound {
		return false, nil
	}
	return blockSize > -1, err
}
```

Reply:
That sounds like the right way to go.

Reply:
@Stebalien This is done. I pushed out these changes and squashed into a single commit.